TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs

```diff
@@ -0,0 +1,177 @@
+		var jsonData = JSON.Parse (data);
+		var games = jsonData ["Games"];
+		for (int i = 0; i < games.Count; i++) {
+			var curr = games [i];
+			if (String.Compare(curr ["game_id"], gameID)==0) {
+				Debug.Log ("I'm in if");
+				// attach popup attributes
+				newPopup.transform.FindChild ("Changing Objects/GameName").GetComponent<Text>().text = curr ["game_name"];
+				newPopup.transform.FindChild ("Changing Objects/Whom").GetComponent<Text>().text = curr ["maker_id"];
+				newPopup.transform.FindChild ("Changing Objects/howManyPart").GetComponent<Text>().text = curr ["participant"];
+				newPopup.transform.FindChild ("Changing Objects/howManyTrea").GetComponent<Text> ().text = curr ["treasure_count"];
+				// assign onClick event to BackButton
+				Transform backButton = newPopup.transform.FindChild("BackButton");
+				Button backB = backButton.GetComponent<Button> ();
+				backB.onClick.AddListener (TurnDownPopup);
+				// assign onClick event to DropOutButton/JoinButton
+				if(isMyGame)
+				{
+					Transform dropOutButton = newPopup.transform.FindChild("DropOutButton");
+					Button dropB = dropOutButton.GetComponent<Button> ();
+					dropB.onClick.AddListener (() => DropOrJoin(true, "gg", curr["game_id"])); // todo: put valid usn instead of "gg"
+				}
+				else
+				{
+					Transform joinButton = newPopup.transform.FindChild ("JoinButton");
+					Button joinB = joinButton.GetComponent<Button> ();
+					joinB.onClick.AddListener (() => DropOrJoin (false, "gg", curr ["game_id"])); // todo: same as above
+				}
+				// attach treasure list
+				// parse treasures
+				var treasures = curr ["Treasures"];
+				// check if there is an error
+				if (curr["treasure_count"].AsInt != treasures.Count) {
+					Debug.Log ("something wrong with treasure_count");
+				}
+				// make treasure objects 
+				for (int j = 0; j < treasures.Count; j++) 
+				{
+					// todo: 
+				}
+			}
+		}
+	}
+
+	public void TurnDownPopup(){
+		Debug.Log ("called");
+		Destroy(GameObject.FindWithTag("Popup"));
+		Debug.Log ("turned down popup");
+	}
+
+	public void DropOrJoin(bool isDrop, string userName, string gameID){
+		// inform server that this user drops out of the game
+		if (!gameObject.GetComponent<NetworkManager> ().enabled) 
+		{
+			Debug.Log ("DropOrJoin called when NetworkManager is not enabled");
+		}
+		else 
+		{
+			string str = "";
+			if (isDrop) 
+			{
+				str = "{\"flag\":16, \"usn\":\"" + userName + "\", \"game_id\":" + gameID + "\"}";
+			} 
+			else 
+			{
+				str = "{\"flag\":14, \"usn\":\"" + userName + "\", \"game_id\":" + gameID + "\"}";
+			}
+			NetworkManager.instance.SendData (str);
+		}
+		// reset data so that if you go back to Field View, treasure setup would start again
+		TreasureSetupController.userGameTreasureData = ""; // needed if we use cache flush
+		// close popup and delete this gamelist from scroll view
+		Destroy(GameObject.Find("Canvas/Scroll View/Viewport/Content/" + gameID));
+		TurnDownPopup();
+	}
+
+	/*
```

Comment:
Uhm... no.... thought that maybe more modulizing would be good but would not gonna use that i guess